python/lsst/daf/butler/cli/cliLog.py

```diff
@@ -273,3 +284,24 @@ def _recordComponentSetting(cls, component):
         component levels."""
         componentSettings = cls.ComponentSettings(component)
         cls._componentSettings.append(componentSettings)
+
+    @classmethod
+    def replayConfigState(cls, configState):
+        """Re-create configuration using configuration state recorded earlier.
+
+        Parameters
+        ----------
+        configState : `list` of `tuple`
+            Tuples contain a method as first item and arguments for the method,
+            in the same format as ``cls.configState``.
+        """
+        if cls._initialized or cls.configState:
+            # Already initialized, do not touch anything.
+            log = logging.getLogger(__name__.partition(".")[2])
+            log.warning("Log is already initialized, will not replay configuration.")
+            return
+
+        # execute each one in order
+        for call in configState:
+            method, *args = call
+            method(*args)
```

Comment:
why not use `functools.partial`? (binding arg vals to funcs is what it's made for)

Reply:
I thought about that but decided that I want to avoid extra imports and complexity. There are certainly cases when `partial` is useful, but in this case it's easy to do things without it.